presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusClient.java

```diff
@@ -174,4 +239,33 @@ private Optional<String> getBearerAuthInfoFromFile()
             }
         });
     }
+
+    public SSLContext getSSLContext()
+            throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException
+    {
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, getTrustManagerFactory().getTrustManagers(), new java.security.SecureRandom());
+        return sslContext;
+    }
+
+    public TrustManagerFactory getTrustManagerFactory()
+            throws KeyStoreException, NoSuchAlgorithmException
```

Comment:
can this throw NoSuchAlgorithmException? It's caught below.

Reply:
TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm()); at line 267 throws NoSuchAlgorithmException

Reply:
perhaps move this into the try block then?

Reply:
Hi @elharo ,
Moved to try block, please review

Reply:
remove from throws clause then.
also consider if you want to convert KeyStoreException to PrestoException here.

Reply:
removed NoSuchAlgorithmException from throws clause.
KeyStoreException is converting to PrestoException on line 221
        catch (KeyStoreException e) {
            throw new PrestoException(KEYSTORE_OPERATION_ERROR, "Keystore operation error", e);
        }

Reply:
Hi @elharo could you review this?

Reply:
hi @elharo could you please continue this review?